src/Evolution/Imex/Protocols/ImplicitSector.hpp

```diff
@@ -0,0 +1,155 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+/// \cond
+class DataVector;
+template <typename X, typename Symm, typename IndexList>
+class Tensor;
+namespace Tags {
+template <typename Tag>
+struct Source;
+}  // namespace Tags
+/// \endcond
+
+namespace imex::protocols {
+/// Protocol for an implicit sector of an IMEX system.
+///
+/// An implicit sector describes the sources for one implicit solve
+/// performed during IMEX evolution.  A system may have multiple
+/// implicit sectors, but they must be independent, i.e., their
+/// sources must not depend on any tensors in other sectors.
+///
+/// Classes implementing this protocol must define:
+///
+/// * a `tensors` type alias of tags for the variables to be solved for
+///
+/// * an `initial_guess` type to be passed to `db::mutate_apply`,
+///   taking additional arguments for the inhomogeneous terms \f$X\f$
+///   and implicit weight \f$w\f$ in the equation to be solved:
+///   \f$u = X + w S(u)\f$.  (See example below.)  It must return a
+///   `std::vector<GuessResult>` indicating, for each point, whether
+///   the implicit equation has been solved analytically or whether
+///   the numerical solve should continue.  An empty return is
+///   equivalent to `GuessResult::InitialGuess` for every point.  When
+///   this is called, the sector variables will have the value from
+///   the explicit part of the time step.  This mutator will not be
+///   called if the implicit weight is zero, as a system-independent
+///   analytic solution is available in that case.  If using the value
+///   of the explicit step as an initial guess is acceptable, this can
+///   be the type `imex::GuessExplicitResult`.
+///
+/// * a `solve_attempts` list of sources that will be attempted to be
+///   solved, in order.  The first one that succeeds at each point
+///   will be used.  Pieces of code that need "the" source for the
+///   sector will use the source from the first entry.  Each attempt
+///   must be a struct with the following types:
+///
+///   * lists used to construct a DataBox during the pointwise
+///     implicit solve:
+///
+///     * `tags_from_evolution` for tags to be made available from the
+///        evolution DataBox.  Volume quantities will be reduced to
+///        have one grid point, with the appropriate value for the
+///        point being solved for.
```

Comment:
Please mention that this excludes the tags from `tensors` (that needs to be solved).